src/sentry/migrations/0049_auto_20200302_1955.py

```diff
@@ -0,0 +1,92 @@
+from __future__ import unicode_literals
+
+from datetime import datetime
+from django.db import migrations
+from django.db.models import OuterRef, Exists
+
+# Date we should cutoff adding the completed alert rule task for orgs (this is
+# after we released the task)
+ALERT_RULE_ADDED_CUTOFF = datetime(2020, 3, 2)
+
+
+def mark_all_projects_completed_alert_rule_onboarding(apps, schema_editor):
+    from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+    from sentry.models import OnboardingTaskStatus, OnboardingTask
+
+    Project = apps.get_model("sentry", "project")
+    OrganizationOnboardingTask = apps.get_model("sentry", "organizationonboardingtask")
+
+    # Locate all projects, annotated which have not completed alert onboarding
+    # tasks.
+    #
+    # Queryplan:
+    #
+    #   Seq Scan on sentry_project (cost=0.00..3989649.04 rows=297362 width=103) (actual time=391.639..890.274 rows=575967 loops=1)
+    #   Filter: ((date_added <= '2020-03-02 00:00:00+00'::timestamp with time zone) AND (NOT (alternatives: SubPlan 3 or hashed SubPlan 4)))
+    #   Rows Removed by Filter: 19720
+    #   SubPlan 1
+    #   -> Index Scan using sentry_organizationonboar_organization_id_47e98e05cae29cf3_uniq on sentry_organizationonboardingtask u0 (cost=0.42..4.45 rows=1 width=0) (never executed)
+    #   Index Cond: ((organization_id = sentry_project.organization_id) AND (task = 10))
+    #   Filter: ((status = 1) AND (project_id = sentry_project.id))
+    #   SubPlan 2
+    #   -> Seq Scan on sentry_organizationonboardingtask u0_1 (cost=0.00..25585.33 rows=20264 width=16) (actual time=0.063..166.531 rows=20947 loops=1)
+    #   Filter: ((status = 1) AND (task = 10))
+    #   Rows Removed by Filter: 874039
+    #   SubPlan 3
+    #   -> Index Scan using sentry_organizationonboar_organization_id_47e98e05cae29cf3_uniq on sentry_organizationonboardingtask u0_2 (cost=0.42..4.45 rows=1 width=0) (never executed)
+    #   Index Cond: ((organization_id = sentry_project.organization_id) AND (task = 10))
+    #   Filter: ((status = 1) AND (project_id = sentry_project.id))
+    #   SubPlan 4
+    #   -> Seq Scan on sentry_organizationonboardingtask u0_3 (cost=0.00..25585.33 rows=20264 width=16) (actual time=0.081..203.465 rows=20947 loops=1)
+    #   Filter: ((status = 1) AND (task = 10))
+    #   Rows Removed by Filter: 874039
+    #   Planning time: 1.811 ms
+    #   Execution time: 913.912 ms
+    #
+    # A range QuerySet wrapper will be used to iterate this query.
+    target_projects = Project.objects.annotate(
+        alert_task_complete=Exists(
+            OrganizationOnboardingTask.objects.filter(
+                organization_id=OuterRef("organization_id"),
+                project_id=OuterRef("id"),
+                task=OnboardingTask.ALERT_RULE,
+                status=OnboardingTaskStatus.COMPLETE,
+            )
+        )
+    ).filter(alert_task_complete=False, date_added__lte=ALERT_RULE_ADDED_CUTOFF)
```

Comment:
Since this takes 3s or so to run, it might actually be more efficient to just iterate through every project, since we're hitting over 90% of them anyway. Otherwise we'll run this every 1000 projects, which will slow down the execution time a bit.

Reply:
Ok, removed the range queryset

Reply:
That's not what he was suggesting. He was suggesting to drop the `filter` and just iterate over all projects with the wrapper instead. Since the query to do that is efficient. This one isn't.

Reply:
Ahh I see, I misunderstood. Okay I brought back the iterator helper and dropped the filter and skip in python.